ompi/runtime/ompi_rte.c

```diff
@@ -777,6 +778,28 @@ int ompi_rte_init(int *pargc, char ***pargv)
         pmix_proc_is_bound = false;
     }
 
+    /* get our numa rank from PMIx */
+    if (pmix_proc_is_bound) {
+        OPAL_MODEX_RECV_VALUE_OPTIONAL(rc, PMIX_NUMA_RANK,
+                                       &pmix_process_info.my_name, &u16ptr, PMIX_UINT16);
+        if (PMIX_SUCCESS != rc) {
+            if (ompi_singleton) {
+                /* just assume the numa_rank is 0 */
+                u16 = 0;
+            } else {
+                ret = opal_pmix_convert_status(rc);
+                error = "numa rank";
+                goto error;
+            }
+        }
+        pmix_process_info.my_numa_rank = u16;
+    } else {
+        /* if processes are not bound, numa_rank is not available
+         * assume numa_rank is 0
+         */
+        pmix_process_info.my_numa_rank = 0;
```

Comment:
I agree with Ralph, only stronger.  We should make this MAX_<datatype size>, not 0.